File: backend/typescript/services/interfaces/storageService.ts
interface IFileStorageService {
  /**
   * Retrieves file
   * @param fileName name of file
   * @returns response
   * @throws Error if file is not retrieved
   */
  getFile(fileName: string): Promise<string>;

  /**
   * Creates file
   * @param fileName name of file
   * @param filePath path of file
   * @throws Error if name of file already exists
   * @throws Error if file is not uploaded
   */
  createFile(fileName: string, filePath: string): Promise<void>;

  /**
   * Updates file
   * @param fileName name of file
   * @param filePath path to file
   * @throws Error if name of file does not exist
   * @throws Error if file is not retrieved
   */
  updateFile(fileName: string, filePath: string): Promise<void>;

  /**
   * Deletes file
   * @param fileName name of file
   * @throws Error if name of file does not exist
   * @throws Error if file is not retrieved
   */
  deleteFile(fileName: string): Promise<void>;
}

export default IFileStorageService;

File: backend/python/app/services/interfaces/storage_service.py
from abc import ABC, abstractclassmethod, abstractmethod

class IStorageService(ABC):
    """ 
    Interface innit
    """

    @abstractmethod
    def get_file(self, file_name):
        pass 

    @abstractmethod
    def create_file(self, file_name, file):
        pass 

    @abstractmethod
    def update_file(self, file_name, file):
        pass 

    @abstractmethod
    def delete_file(self, file_name):
        pass 

